lgn-provers/src/provers/v1/query/task.rs

```diff
@@ -297,18 +299,55 @@ impl<P: StorageQueryProver> Querying<P>
             },
             QueryStep::Revelation(rev) =>
             {
-                return self
-                    .prover
-                    .prove_revelation(
-                        &pis,
-                        rev.placeholders
-                            .clone()
-                            .into(),
-                        rev.query_proof
-                            .to_owned(),
-                        rev.indexing_proof
-                            .to_owned(),
-                    );
+                match rev
+                {
+                    RevelationInput::Aggregated {
+                        placeholders,
+                        indexing_proof,
+                        query_proof,
+                        ..
+                    } =>
+                    {
+                        return self
+                            .prover
+                            .prove_aggregated_revelation(
+                                &pis,
+                                placeholders
+                                    .clone()
+                                    .into(),
+                                query_proof.clone_proof(),
+                                indexing_proof.clone_proof(),
+                            );
+                    },
+                    RevelationInput::Tabular {
+                        placeholders,
+                        indexing_proof: preprocessing_proof,
+                        matching_rows,
+                        column_ids,
+                        limit,
+                        offset,
+                        ..
+                    } =>
+                    {
+                        return self
+                            .prover
+                            .prove_tabular_revelation(
+                                &pis,
+                                placeholders
+                                    .clone()
+                                    .into(),
+                                preprocessing_proof.clone_proof(),
+                                matching_rows
+                                    .into_iter()
+                                    .cloned()
+                                    .map(HydratableMatchingRow::into_matching_row)
```

Comment:
Ok i see the cloning there, not on QE then right ? 
Couldn't just pass the value to the new struct expected by mp2 ? Why do we need to clone it ?

Reply:
Cf. above; we should have refs or CoW for all these IMHO, but I'd prefer to do it in another PR.

Reply:
FTR, this is a temporary solution until hydration is dropped altogether for the queries.